code/modules/mining/machine_processing.dm

```diff
@@ -1,429 +1,587 @@
+
+		send_signal(list("toggle_alloying" = ore_id))
+		request_status()
+		return 1
+
+	if(href_list["toggle_power"])
+		send_signal(list("toggle_power" = 1))
+		request_status()
+		return 1
+
+	if(href_list["toggle_ores"])
+		show_all_ores = !show_all_ores
+		return 1
+
+	if(href_list["eject"])
+		id.forceMove(get_turf(src))
+		if(!usr.get_active_hand()) //Nothing in the user's hand.
+			usr.put_in_hands(id)
+
+		id = null
+		updateUsrDialog()
+		return 1
+
+	if(href_list["claim"])
+		send_signal(list("claimcredits" = get_card_account(id)))
+		request_status()
+		return 1
+
+	if(href_list["insert"])
+		if(id)
+			usr << "<span class='notify'>There is already an ID in the console!</span>"
+			return 1
+		var/obj/item/weapon/card/id/I = usr.get_active_hand()
+		if(istype(I))
+			usr.drop_item(I, src)
+			id = I
+		updateUsrDialog()
+		return 1
+
+/obj/machinery/computer/smelting/attackby(var/obj/item/W, var/mob/user)
+	if(istype(W, /obj/item/weapon/card/id))
+		if(id)
+			usr << "<span class='notify'>There is already an ID in the console!</span>"
+			return 1
+
+		var/obj/item/weapon/card/id/I = W
+		usr.drop_item(I, src)
+		id = I
+		updateUsrDialog()
+		return 1
+
+	. = ..()
+
+//Just a little helper proc
+/obj/machinery/computer/smelting/proc/send_signal(list/data)
+	if(!frequency)
+		return
+
+	var/datum/signal/signal = new
+	signal.data["tag"] = smelter_tag
+	signal.transmission_method = 1 //radio signal
+	signal.source = src
+	signal.data += data
+
+	radio_connection.post_signal(src, signal)
+
+/obj/machinery/computer/smelting/receive_signal(datum/signal/signal)
+	if(stat & (NOPOWER | BROKEN) || !signal || !signal.data["tag"] || signal.data["tag"] != smelter_tag)
+		return
+
+	smelter_data = signal.data
+	updateUsrDialog()
+
+/obj/machinery/computer/smelting/proc/request_status()
+	smelter_data = null
+	send_signal(list("sigtype" = "status"))
+
+/obj/machinery/computer/smelting/multitool_menu(var/mob/user, var/obj/item/device/multitool/P)
+	return {"
+	<ul>
+		<li><b>Frequency:</b> <a href="?src=\ref[src];set_freq=-1">[format_frequency(frequency)] GHz</a> (<a href="?src=\ref[src];set_freq=[1439]">Reset</a>)</li>
+		<li>[format_tag("ID Tag","smelter_tag")]</li>
+	</ul>
+	"}
 
 /**********************Mineral processing unit**************************/
 
 /obj/machinery/mineral/processing_unit
-	name = "furnace"
+	name = "Ore Processor"
 	icon = 'icons/obj/machines/mining_machines.dmi'
 	icon_state = "furnace"
 	density = 1
-	anchored = 1.0
-	var/obj/machinery/mineral/input = null
-	var/obj/machinery/mineral/output = null
-	var/obj/machinery/mineral/CONSOLE = null
+	anchored = 1
+	idle_power_usage = 50
+	active_power_usage = 500 // This shit's able to compress tiny little diamonds into really big diamonds, of course this uses a lot of power.
+	machine_flags = SCREWTOGGLE | CROWDESTROY | MULTITOOL_MENU
+	light_power_on = 2
+	light_range_on = 3
+	light_color = LIGHT_COLOR_ORANGE
+
+	var/frequency = 1367// Same as conveyors
```

Comment:
oh you and your magic numbers, LITERALLY EVERYTHING IN THE HISTORY OF SIGNALS HASN'T USED A DEFINE FOR FREQUENCIES